### PR TITLE
Allow to give admin password via secret

### PR DESCRIPTION
**Pull request by sylvainOL**
Instead of setting the admin password via a parameter in the helm chart,
provide the ability to give it via an existing secret.

### CODE REVIEW

**Comment by sylvainOL** [comment]
Hello, I'm sorry to bother you but is there any reason not to at least launch tests? (same question for #1290 and #1291)

**Comment by sylvainOL** [comment]
@zyyw , @MinerYang can you take a look?

File: README.md [hunk 1]
```diff
@@ -196,6 +196,8 @@ The following table lists the configurable parameters of the Harbor chart and th
...
 | `harborAdminPassword`                                                | The initial password of Harbor admin. Change it from portal after launching Harbor                                                                                                                                                                                                                                                                                                                                                                                | `Harbor12345`                         |
+| `existingSecretAdminPassword`                                                | The name of secret whera admin password can be found. Harbor                                                                                                                                                                                                                                                                                                                                                                                |                          |
```
Comment (pitwegner) [inline README.md:199]: typo in "where". Also, why the trailing "Harbor"? Same in line below.

File: templates/core/core-dpl.yaml [hunk 2]
```diff
@@ -86,7 +86,14 @@ spec:
               secretKeyRef:
                 name: "{{ template "harbor.jobservice" . }}"
                 key: JOBSERVICE_SECRET
-         {{- if .Values.internalTLS.enabled }}
+          {{- if .Values.existingSecretAdminPassword }}
```
Comment (pitwegner) [inline templates/core/core-dpl.yaml:89]: why not simply add another if-guarded line to the secret? This kind of breaks the convention here.

File: templates/exporter/exporter-dpl.yaml [hunk 3]
```diff
@@ -64,6 +64,13 @@ spec:
               name: {{ .Values.database.external.existingSecret }}
               key: password
         {{- end }}
+        {{- if .Values.existingSecretAdminPassword }}
```
Comment (pitwegner) [inline templates/exporter/exporter-dpl.yaml:67]: Same as above. Keep it simple by adding a line to the existing secret.

**Review by pitwegner — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sylvainOL) [reply README.md:199]: good catch thanks, I've just resolved it!

File: templates/core/core-dpl.yaml (hunk 2 above)
Comment (sylvainOL) [reply templates/core/core-dpl.yaml:89]: Maybe I don't catch your comment.

But if the comment is to set it in "core" secret that is generated, I guess we can't
The goal here is to retrieve the admin password value from a "previously generated" one.
Then, we cannot add it to the core secret.
And then we need to set it this way AFAIK.

please don't hesitate to correct me as I may didn't understood your comment

File: templates/exporter/exporter-dpl.yaml (hunk 3 above)
Comment (sylvainOL) [reply templates/exporter/exporter-dpl.yaml:67]: same as previous comment

**Review by sylvainOL — COMMENTED** [review]
Thanks for the review!

I solved readme issue and tried to answer on the secret part.

File: templates/core/core-dpl.yaml (hunk 2 above)
Comment (pitwegner) [reply templates/core/core-dpl.yaml:89]: Absolutely right. My fault.

**Comment by rdelpret** [comment]
@pitwegner any movement on this? We would really like this feature as well.

**Comment by pitwegner** [comment]
@rdelpret I'm not related to the goharbor project at all, I just added my 2 cents to the PR because I'm interested in the feature as well

**Comment by rdelpret** [comment]
@MinerYang @zyyw Any movement on this? also looking for any help with maintaining this chart?

**Comment by zyyw** [comment]
closing it and then reopen it to run the ci

**Review by zyyw — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**